packages/datadog/src/index.ts

```diff
@@ -59,6 +50,20 @@ export class DataDogMiddleware implements Middleware {
     tracer.inject(datadogSpan, this.DATADOG_CONTEXT_FORMAT, contextData);
     return contextData;
   }
+
+  private createBaseTraceOptionsFromContext(
+    context: MiddlewareContext
+  ): tracer.TraceOptions & SpanOptions {
+    let tags = {};
+    if (typeof context.payload !== 'string') {
+      tags = { ...context.payload };
+    }
+
+    return {
+      resource: context.topic,
+      tags,
+    };
+  }
```

Comment:
Nitpick, but you can simplify this function like so:
```suggestion
  private createBaseTraceOptionsFromContext(
    context: MiddlewareContext
  ): tracer.TraceOptions & SpanOptions {
    return {
      resource: context.topic,
      tags: typeof context.payload !== "string" ? { ...context.payload } : {},
    };
  }
```

Reply:
i agree on the nitpick, but would add a little more meaning to the test value:

```
 private createBaseTraceOptionsFromContext(
	    context: MiddlewareContext
	  ): tracer.TraceOptions & SpanOptions {
  	  const validPayload = typeof context.payload !== "string";
  	  const tags = validPayload ? { ...context.payload } : {};
  	  
	    return {
	      resource: context.topic,
	      tags,
	    };
	  }
```

this makes debugging and logging so much easier than mashing everything into one line, and provides context as to what the condition is actually intending to test

Reply:
Done